### PR TITLE
Location property

### PR DESCRIPTION
**Pull request by ericshermancs**

### CODE REVIEW

**Comment by Huddie** [comment]
@ericshermancs this pr is just fixing location ya?

**Comment by ericshermancs** [comment]
may accidentally be continuation of class search, but other than that yes

**Comment by Huddie** [comment]
Can you get rid of the suff unrelated to the location (remove class search stuff)

**Comment by ericshermancs** [comment]
not sure how

**Comment by Huddie** [comment]
Youll prob have to create a clean branch and redo. 

`git checkout -b <branch_name> upstream/master`

**Comment by ericshermancs** [comment]
want to just review the class search on the separate branch i made for that?

File: cunyfirstapi/tests/tests.py [hunk 1]
```diff
@@ -10,11 +10,24 @@
...
 import unittest
+from ..locations_enum import Location
+from ..cunyfirstapi import CUNYFirstAPI
 
 class TestTest(unittest.TestCase):
     def test(self):
         self.assertEqual(True, True)
 
+
+class ClassSearchTest(unittest.TestCase):
```
Comment (Huddie) [inline cunyfirstapi/tests/tests.py]: This isnt gonna work. 

1. Add the argsParse option to main, store a global var for username and password and use those gl

**Comment by Huddie** [comment]
I cant merge the code twice. It just creates a long confusing commit history. And reverting becomes difficult.

**Comment by ericshermancs** [comment]
review it, and if its good, close the PR, and merge this one

**Comment by Huddie** [comment]
Danger is failing. there is also a new version of it that needs to be added. So make sure you pull the upstream/master

File: cunyfirstapi/transcript.py [hunk 2]
```diff
@@ -60,29 +60,28 @@ def move(self):
...
 
-        college_code = self._college_code
+        college_code = self._location._college_code
 
         if alt_college_code:
             college_code = alt_college_code
 
         url = constants.CUNY_FIRST_TRANSCRIPT_REQUEST_URL
-        r = self._session.get(url)
+        r = self._location_session.get(url)
```
Comment (Huddie) [inline cunyfirstapi/transcript.py]: location._session ya?

File: cunyfirstapi/transcript.py [hunk 3]
```diff
@@ -93,22 +92,22 @@ def download(self, alt_college_code=None):
         data['ICYPos'] ='115'
 
         # tell it we picked that college
-        r = self._session.post(url, data=data)
+        r = self._location._session.post(url, data=data)
```
Comment (Huddie) [inline cunyfirstapi/transcript.py]: Make a get method for location's session if your going to call it often

File: cunyfirstapi/tests/tests.py [hunk 4]
```diff
@@ -10,15 +10,45 @@
...
+            subject = 'CSCI', course_number='111')
+        self.assertTrue(len(results['results']) > 0)
+
+
+
 def run_test():
     unittest.main()
 
 def main():
+	global username
```
Comment (Huddie) [inline cunyfirstapi/tests/tests.py]: Are these properly indented?

File: cunyfirstapi/login.py [hunk 5]
```diff
@@ -74,6 +74,11 @@ def login(username, password):
         data=data
     )
     response = new_session.get(constants.CUNY_FIRST_SIGNED_IN_STUDENT_CENTER_URL)
+
+    tree = html.fromstring(response.text)
+
```
Comment (Huddie) [inline cunyfirstapi/login.py]: remove blank line

File: cunyfirstapi/locations_enum.py [hunk 6]
```diff
@@ -13,22 +13,37 @@
...
     student_grades = 1
     transcript     = 2
+    class_search   = 3
 
     def __init__(self, session, college_code):
         self._college_code = college_code
         self._session = session
 
     def get_location_object(self, location):
+        '''
```
Comment (Huddie) [inline cunyfirstapi/locations_enum.py]: Remove this code

File: cunyfirstapi/class_search.py [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+                    'meeting_dates' : ''.join(row.xpath('.//span[contains(@id,"MTG_TOPIC$")]/text()')).strip(),
+                    'status' : re.search(r'OPEN|CLOSED|WAITLIST',''.join(row.xpath('.//img[@class="SSSIMAGECENTER"]/@src'))).group(0).strip(),
+                    'bookstore_link' : '',#''.join(row.xpath('.//a[contains(@id,"MTG_INSTR$")]/text()'))
+                    'mode_of_instruction' : ''.join(row.xpath('.//span[contains(@id,"INSTRUCT_MODE_DESCR$")]/text()')).strip()
+                }
+
+                #pprint(row_info)
+                result.append(row_info)
+
+        return {'results': result,
```
Comment (Huddie) [inline cunyfirstapi/class_search.py]: Fix this style

**Review by Huddie — CHANGES_REQUESTED** [review]
I had review comment on the other PR. Make sure to review those as well.

File: cunyfirstapi/transcript.py (hunk 2 above)
Comment (ericshermancs) [reply cunyfirstapi/transcript.py]: _location, this is the location property of the action object

File: cunyfirstapi/transcript.py (hunk 3 above)
Comment (ericshermancs) [reply cunyfirstapi/transcript.py]: already exists, I will substitute it

File: cunyfirstapi/tests/tests.py (hunk 4 above)
Comment (ericshermancs) [reply cunyfirstapi/tests/tests.py]: i will check

File: cunyfirstapi/transcript.py (hunk 2 above)
Comment (Huddie) [reply cunyfirstapi/transcript.py]: your missing the `.`

File: cunyfirstapi/class_search.py [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+
+        
+        result = []
+        if re.search(r'<span  class=\'SSSMSGWARNINGTEXT\'.*</span>', response.text):
+        	#print(re.search(r'<span  class=\'SSSMSGWARNINGTEXT\'.*</span>', response.text).group(0))
+        	tree = html.fromstring(re.search(r'<span  class=\'SSSMSGWARNINGTEXT\'.*</span>', response.text).group(0))
+        	warning = ''.join(tree.xpath('//span[@class="SSSMSGWARNINGTEXT"]/text()'))
+        	return {
+        		'results': result,
+        		'success' : False,
```
[the remark this answers is not in the document]
Comment (Huddie) [reply cunyfirstapi/class_search.py]: ^^

File: cunyfirstapi/transcript.py [hunk 9]
```diff
@@ -93,22 +92,22 @@ def download(self, alt_college_code=None):
         data['ICYPos'] ='115'
 
         # tell it we picked that college
-        r = self._session.post(url, data=data)
+        r = self.location()_session.post(url, data=data)
```
[the remark this answers is not in the document]
Comment (Huddie) [reply cunyfirstapi/transcript.py]: ^^ invalid synatx. also add a get method for session. self._location was fine. 
`self._location.get_session()` or self.location().get_session()
or `self.location().session()` or similar

**Pull request closed without merging**